internal/liveness/metrics.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2020 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package liveness
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ceph/ceph-csi/internal/rbd"
+	"github.com/ceph/ceph-csi/internal/util"
+
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+var (
+	rbdMapping = *prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "csi",
+		Name:      "ceph_rbd",
+		Help:      "ceph rbd image pvc mapping"},
+		[]string{"pvc_name", "pvc_namespace", "pv_name", "pool", "imageID"})
+	cephfsMapping = *prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "csi",
+		Name:      "ceph_fs",
+		Help:      "cephfs subvolume pvc mapping"},
+		[]string{"pvc_name", "pvc_namespace", "pv_name", "volume_name", "subvolume_group", "subvolume_name"})
+)
+
+func getMapping(timeout time.Duration, client *k8s.Clientset, numGoroutines int) {
+	klog.V(4).Infof("Start recording mapping")
+	pvcs, err := client.CoreV1().PersistentVolumeClaims("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Get pvclist failed: %v", err)
+		return
+	}
+	driverName := os.Getenv("DRIVER_NAME")
```

Comment:
why from ENV, make an RPC call to CSI driver and get the plugin Info?

Reply:
done